File: DSA/AnshulJain_590014934/Day_41_Question_3.py
num = input("Enter a number: ")#Take the input from the user 
k = int(input("Enter the number of digits to remove: "))
stack = []
for digit in num:
    while k and stack and stack[-1] > digit:
        stack.pop()
        k -= 1
    stack.append(digit)
while k:
    stack.pop()
    k -= 1
result = ''.join(stack).lstrip('0')
print(result if result else '0')
#for example:
# Input: "1432219", k = 3
# Output: "1219"

File: DSA/NiharikaChand_590017697/Day41Question3.java
import java.util.*;
class Day41Question3
{
    public static String remove(String num, int k) 
    {
        Stack<Character> stack = new Stack<>();
        for (int i = 0; i < num.length(); i++) 
        {
            char ch = num.charAt(i);
            while (!stack.isEmpty() && k > 0 && stack.peek() > ch) 
            {
                stack.pop();
                k--;
            }
            stack.push(ch);
        }

        while (k > 0 && !stack.isEmpty()) 
        {
            stack.pop();
            k--;
        }

        StringBuilder result = new StringBuilder();
        for (int i = 0; i < stack.size(); i++) 
        {
            result.append(stack.get(i));
        }

        while (result.length() > 1 && result.charAt(0) == '0') 
        {
            result.deleteCharAt(0);
        }
        if (result.length() == 0) 
        {
            return "0";
        }   
        else 
        {
            return result.toString();
        }
    }

    public static void main(String args[]) 
    {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a number: ");
        String num = sc.nextLine();

        System.out.print("Enter k (digits to remove): ");
        int k = sc.nextInt();

        String result = remove(num, k);
        System.out.println("Smallest number: " + result);
        sc.close();
    }
}

